woocommerce/woocommerce
# [LYS] Comin soon block - Refactor and hide Styles block on Store Only

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR cleans up the Coming soon block by generating the CSS in a separate function, creating a proper Save function, and setting the API version to 2 to eliminate warnings.

It also removes the `<styles>` block found in the Store only template by placing all contents inside the Coming Soon block.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. In a fresh JN site make sure the feature flag `launch-your-store` is turned on.
2. Visit the template to make sure it loads correctly Appearance > Editor > Templates > Page: Coming soon.
3. Visit the front end Coming soon by visiting the site in incognito mode, it should render correctly.
4. Change the settings to Store Only. WooCommerce > Settings > Site Visibility > "Restrict to store pages only" toggle ON.
5. Visit the template once again. Notice there is no code block at the bottom.
6. Visit the front end Coming soon once again in incognito mode. This time you'll need to visit a shop page.
7. Change the theme to Storefront (or any other non-FSE theme).
8. Repeat step 6, it should render correctly as well.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 
update to previously unreleased feature
</details>
